include/onnxruntime/core/session/onnxruntime_c_api.h

```diff
@@ -385,6 +386,18 @@ struct OrtApi {
   OrtStatus*(ORT_API_CALL* SessionGetOverridableInitializerName)(_In_ const OrtSession* sess, size_t index,
                                                                  _Inout_ OrtAllocator* allocator, _Outptr_ char** value)NO_EXCEPTION;
 
+  /**
+   * \param out is set to a null terminated string allocated using 'allocator'. The caller is responsible to free it.
+   */
+  OrtStatus*(ORT_API_CALL* SessionEndProfiling)(_In_ OrtSession* sess, _Inout_ OrtAllocator* allocator,
+                                                _Out_ char* out)NO_EXCEPTION;
```

Comment:
Seems a little unbalanced to have an 'end' method without a 'start' method. I'm assuming that if I call EnableProfiling it will start automatically, hence a way to stop that is required, but wondering if DisableProfiling can be used for that.

Reply:
Just a reminder (it came up in our scrum), but all new functions need to be at the end of the structure to preserve backwards compatibility.

Reply:
is disabling profiling the same thing as "ending" an existing profile that is running though ? Initially, I found it a bit confusing too - took me a while to figure out that profiling is started automatically when enabled. For now, I chose to follow the same pattern as the Python APIs...